### PR TITLE
State distribution and Summary end points for Title I

### PR DESCRIPTION
**Pull request by ywkim312**
**Three new endpoints were added**
http://localhost:5000/pdl/titles/title-i/state-distribution
http://localhost:5000/pdl/titles/title-i/summary
http://localhost:5000/pdl/titles/title-ii/programs/eqip-ira/practice-names

**State distribution data can be verified by the following query (change year and state)**
```
SELECT
    SUM(payment) AS totalPaymentInDollars
FROM
    pdl.payments
WHERE
    state_code = 'IL'
    AND year = 2014;
```

**Summary data can be verified by the following queries**

get the whole title I (title id = 100) between 2014 to 2021
```
SELECT
    SUM(payment) AS totalPaymentInDollars
FROM
    pdl.payments
WHERE
    title_id = 100
    AND year BETWEEN 2014 AND 2021;
```

get the whole subtitle 100 between 2014 to 2021
```
SELECT
    SUM(payment) AS totalPaymentInDollars
FROM
    pdl.payments
WHERE
    subtitle_id = 100
    AND year BETWEEN 2014 AND 2021;
```

### CODE REVIEW

**Review by pengyin-shan — APPROVED** [review]
Looks good to me. Thanks for all the updates @ywkim312 !

File: app/controllers/pdl.py [hunk 1]
```diff
@@ -194,6 +196,24 @@ def allprograms_search():
         return data_json
 
 
+# /pdl/titles/title-i/map:
```
Comment (sandeep-ps) [inline app/controllers/pdl.py:199]: Comment is different from the endpoint.

File: app/pdl.yaml [hunk 2]
```diff
@@ -431,6 +471,26 @@ paths:
           description: Unauthorized
         500:
           description: Internal error
+  /pdl/titles/title-ii/programs/eqip-ira/category:
```
Comment (sandeep-ps) [inline app/pdl.yaml:474]: It may be better to call the endpoint as /practices or /practice-names.

Practice categories are different entities.

**Review by sandeep-ps — APPROVED** [review]
I left a few comments. Approving with the same feedback. The rest looks good to me in brief testing using Postman. Thanks.

File: app/controllers/pdl.py (hunk 1 above)
Comment (ywkim312) [reply app/controllers/pdl.py]: sorry, fixed

File: app/pdl.yaml (hunk 2 above)
Comment (ywkim312) [reply app/pdl.yaml]: Changed to practice-names

**Pull request merged**